types/scheduler.d.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Task priorities that determine the order in which tasks run.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Prioritized_Task_Scheduling_API#task_priorities)
+ */
+export type TaskPriority = 'user-blocking' | 'user-visible' | 'background';
+
+/**
+ * {@link Scheduler.postTask} options.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Scheduler/postTask#options)
+ */
+export type SchedulerPostTaskOptions = {
+  /** The immutable {@link TaskPriority} of the task. One of `"user-blocking"`, `"user-visible"`, or `"background"`. If set, this priority is used for the lifetime of the task and priority set on the `signal` is ignored. */
+  priority?: TaskPriority;
+  /** An {@link AbortSignal} or {@link TaskSignal} that can be used to abort or re-prioritize the task (from its associated controller). */
+  signal?: AbortSignal | TaskSignal;
+  /** The minimum amount of time after which the task will be added to the scheduler queue, in whole milliseconds. The actual delay may be higher than specified, but will not be less. The default delay is 0. */
+  delay?: number;
+};
+
+/**
+ * {@link Scheduler.yield} options.
+ *
+ * [Explainer reference](https://github.com/WICG/scheduling-apis/blob/main/explainers/yield-and-continuation.md#controlling-continuation-priority-and-abort-behavior)
+ */
+export type SchedulerYieldOptions = {
+  /** The priority for the continuation, either a {@link TaskPriority} (`"user-blocking"`, `"user-visible"`, or `"background"`) or `"inherit"` to infer the priority from the current context. */
+  priority?: TaskPriority | 'inherit';
+  /** An {@link AbortSignal} or {@link TaskSignal} that can be used to abort or re-prioritize the task (from its associated controller), or `"inherit"` to inherit the current task's signal. */
+  signal?: AbortSignal | TaskSignal | 'inherit';
+};
```

Comment:
probably the biggest thing to double check because AFAIK only defined via examples in the explainer and in the parameter checking in the polyfill

Reply:
Oh nice, didn't think to check in chromium!